### PR TITLE
Refactor `createEntity` GQL mutation to enable link entity creation

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR refactors the `createEntity` GQL mutation to accept an optional `linkMetadata` parameter which lets callers create link entities.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203301853897838/1203438053318775/f) _(internal)_

## 🔍 What does this change?

See commits.

## 📜 Does this require a change to the docs?

The GQL docs have been updated.

## 🛡 What tests cover this?

None.

## ❓ How to test this?

1. Follow the instructions to run the HASH App using `yarn dev`
2. Log in to the frontend
3. Go to the GQL playground (at localhost:5001/graphql)
4. Create a link entity using `createEntity`. For example to create an org membership:

```GraphQL
mutation createOrgMembershipQuery($properties: JSONObject!, $linkMetadata: LinkMetadata!) {
  createEntity(
    entityTypeId: "http://localhost:3000/@example-org/types/entity-type/org-membership/v/1"
    properties: $properties
    linkMetadata: $linkMetadata
  )
}
```

```JSON
{
  "properties": {
    "http://localhost:3000/@example-org/types/property-type/responsibility/": "Member"
  },
  "linkMetadata": {
    "leftEntityId": "<userEntityId>",
    "rightEntityId": "<orgEntityId>"
  }
}
```

### CODE REVIEW

File: packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts [hunk 1]
```diff
@@ -33,15 +37,43 @@ export const createEntity: ResolverFn<
...
+    entityModel = await LinkEntityModel.createLinkEntity(graphApi, {
+      leftEntityModel,
+      leftOrder: leftOrder ?? undefined,
+      rightEntityModel,
+      rightOrder: rightOrder ?? undefined,
+      properties,
+      linkEntityTypeModel,
+      ownedById: ownedById ?? userModel.getEntityUuid(),
+      actorId: userModel.getEntityUuid(),
+    });
```
Comment (Alfred-Mountfield) [inline packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts:65]: Just something to think about when reworking these model classes soon. I definitely don't think we need to be querying the endpoint entities just to create the thing, we're not even using this information from what I can see, just passing it in to immediately drop it again

**Review by Alfred-Mountfield — APPROVED** [review]
Cheers Ben, looks good!

File: packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts [hunk 2]
```diff
@@ -17,6 +17,13 @@ export const entityTypedef = gql`
     entity: EntityDefinition!
   }
 
+  input LinkEntityMetadata {
```
Comment (Alfred-Mountfield) [inline packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts:20]: Do we want this btw? Or do we want to be consistent with the others like `EntityMetadata` and just use the scalar?

File: packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts (hunk 1 above)
Comment (benwerner01) [reply packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts:65]: Agreed

File: packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts (hunk 2 above)
Comment (benwerner01) [reply packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts]: Ah I didn't think of using scalars as inputs, much prefer this and have removed the type in https://github.com/hashintel/hash/pull/1510/commits/08dff0ed682e439cb2512b14d4910d3dbda154bf

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**